### PR TITLE
Optimize attention kernel v2 1.0, use Gemm replace GemmStridedBatch

### LINKED ISSUE #1133: Remove `W_out_contiguous` and corresponding memory allocation

### PR DESCRIPTION
**Pull request by xinhaoc**
**Description of changes:**

**Related Issues:**

Linked Issues:
- Issue #1133 

Issues closed by this PR:
- Closes #1133 

- - -
This change is [Reviewable](https://reviewable.io/reviews/flexflow/FlexFlow/1190)

### CODE REVIEW

**Comment by xinhaoc** [comment]
#1133

File: tests/inference/python_test_configs/generate_configs.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
 
 # Generate incremental decoding configs
-all_models = llama_models + opt_models + falcon_models + mpt_models
+all_models = llama_models + opt_models + mpt_models
```
Comment (jiazhihao) [inline tests/inference/python_test_configs/generate_configs.py:68]: Any reason we disabled the falcon model tests?

**Review by jiazhihao — APPROVED** [review]
Looks great to me!

File: tests/inference/python_test_configs/generate_configs.py (hunk 1 above)
Comment (xinhaoc) [reply tests/inference/python_test_configs/generate_configs.py:68]: falcon 7B full precision memory inefficient, I'll let it run half precision before merging.

**Pull request merged**